### DISCUSSION TITLE
Which filter/weighting is used for fan-beam reconstruction

### DISCUSSION THREAD

**Discussion by rdilz** [astra-toolbox/astra-toolbox#395] [Q&A]
I am trying to find out what manipulations take place in the "Filtering" part of FBP. Basically, I would like to know all manipulations that take place on a sinogram before the backprojection.

In the case of parallel-beam all that is done is a convolution with a function such as in Eq. 61 of https://engineering.purdue.edu/~malcolm/pct/CTI_Ch03.pdf
I tried this and it works, I get exactly the same reconstruction with my own filtering step + Astra's BP as Astra's FBP.

However, for fan-beam things are more complicated. On pages 91/92 of https://engineering.purdue.edu/~malcolm/pct/CTI_Ch03.pdf the steps required for filtering are explained for fan-beam. However, when I implement these steps, I get results that are slightly different. Less noisy, and off by a few percent (depending on source/detector distance).

From the source code I am not able to find what the exact steps are between a sinogram and the backprojection for fanflat FBP. I would greatly appreciate it if someone could enlighten me.

**Comment by askorikov** [comment]
You could inspect the relevant code in this function: https://github.com/astra-toolbox/astra-toolbox/blob/fe26d706ad217745f8a4f999d434be6eea45d21e/cuda/2d/fbp.cu#L257
